runtime/gcchk/CMakeLists.txt

```diff
@@ -29,52 +29,30 @@ j9vm_add_library(j9gcchk SHARED
 	${gcchk_sources}
 )
 
-if(DEFINED OMR_OS_AIX AND DEFINED CMAKE_C_COMPILER_IS_OPENXL)
-	target_link_libraries(j9gcchk
-		PRIVATE
-			j9vm_interface
-			j9util
-			j9utilcore
-			j9avl
-			j9hashtable
-			j9thr
-			j9stackmap
-			j9pool
-			j9hookable
-			j9modronstartup
-
-			#TODO do we just want to link against j9gc?
-			j9gcbase
-			j9gcstructs
-			j9gcstats
-			j9gccheck
-			omrgc
-			j9gctrc
-			j9gctrcstandard
-			j9gctrcvlhgc
-			j9gc
-	)
-else()
-	target_link_libraries(j9gcchk
-		PRIVATE
-			j9vm_interface
-			j9util
-			j9utilcore
-			j9avl
-			j9hashtable
-			j9thr
-			j9stackmap
-			j9pool
-			j9hookable
+target_link_libraries(j9gcchk
+	PRIVATE
+		j9vm_interface
+		j9util
+		j9utilcore
+		j9avl
+		j9hashtable
+		j9thr
+		j9stackmap
+		j9pool
+		j9hookable
+		j9modronstartup
 
-			#TODO do we just want to link against j9gc?
-			j9gcbase
-			j9gcstructs
-			j9gcstats
-			j9gccheck
-			omrgc
-	)
-endif()
+		#TODO do we just want to link against j9gc?
+		j9gcbase
+		j9gcstructs
+		j9gcstats
+		j9gccheck
+		omrgc
+		j9gctrc
+		j9gctrcstandard
+		j9gctrcvlhgc
+		j9gc
+)
```

Comment:
This is going to change what happens when `!defined(DEFINED OMR_OS_AIX AND DEFINED CMAKE_C_COMPILER_IS_OPENXL)`; why are we modifying non AIX+OPENXL platforms?

Reply:
I believe the original intent was to limit this to AIX and OpenXL, but that condition was removed as requested on Ishita's old PR: https://github.com/eclipse-openj9/openj9/pull/20045#discussion_r1831592007

Reply:
I see; in that case, that done in another PR then. This PR should be solely focused on modifications to the AIX build config to support both xlC and OpenXL.

Reply:
Just opened a new PR for this fix here: https://github.com/eclipse-openj9/openj9/pull/20799